### PR TITLE
FHAC-368:Fix Mimemessage sender for Direct Outbound XDR Messages

### PR DESCRIPTION
**Pull request by alameluchidambaram**
Retrieve the sender from Mime Message "From" header when the MessageMonitoring is enabled for the Outbound XDR.

### CODE REVIEW

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/MimeMessageBuilder.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public MimeMessage build() {
 
         final MimeMessage message = new MimeMessage(session);
 
-        try {
+        try {
```
Comment (christophermay07) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/MimeMessageBuilder.java]: Remove whitespace

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/impl/MessageMonitoringAPI.java [hunk 2]
```diff
@@ -694,4 +687,13 @@ private void deleteElapsedArchivedMessage(MonitoredMessage trackMessage) {
         }
     }
 
+    private String getSenderEmailId(MimeMessage message) throws MessagingException {
+        InternetAddress sender = (InternetAddress) message.getSender();
+        if (sender == null) {
+            InternetAddress[] fromAddresses = (InternetAddress[]) message.getFrom();
+            sender = fromAddresses[0];
```
Comment (christophermay07) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/impl/MessageMonitoringAPI.java]: Has "message" been null checked before this point?  Also, is there a guarantee that the fromAddresses length will be greater than 0?

Comment (alameluchidambaram) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/impl/MessageMonitoringAPI.java]: If the MimeMessage is "Null" it was already caught and exception was thrown by DirectSenderImpl sendOutboundDirect() method before reaching here .

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Comment by nareshcgi** [comment]
Good with the PR.

**Pull request merged**